File: src/com/quotegraph/controller/CoordinatesCalculatorColumns.java
package com.quotegraph.controller;

import com.quotegraph.model.DataLoader;

/**
 * Coordinates calculator for column graph.
 *
 * @author Daniel Peters
 * @version 1.0
 */
public class CoordinatesCalculatorColumns extends CoordinatesCalculator {

  /**
   * Default constructor.
   *
   * @param loader data loader instance reference
   */
  public CoordinatesCalculatorColumns(DataLoader loader) {

    super(loader);

  }

  @Override
  public double translateToPanel(
      double panelSide, double value, double max, double min, boolean inverted) {

    double ratio = (panelSide - 2 * margin) / (min - max);
    double temp = (value - max) * ratio;

    return Math.round(inverted ? panelSide - temp - margin : temp + margin);
  }

}

File: src/main/kotlin/ch/peters/daniel/quotegraph/controller/CoordinatesCalculatorColumns.kt
package ch.peters.daniel.quotegraph.controller

import ch.peters.daniel.quotegraph.model.DayQuote

/**
 * Coordinates calculator for column graph.
 *
 * @author Daniel Peters
 * @version 1.0
 */
class CoordinatesCalculatorColumns(data: List<DayQuote>) : AbstractCoordinatesCalculator(data) {
  override fun translateToPanel(side: Double, value: Double, max: Double, min: Double, inverted: Boolean): Double {
    val ratio = (side - 2 * margin) / (min - max)
    val temp = (value - max) * ratio
    return if (inverted) side - temp - margin else temp + margin
  }
}
